### PR TITLE
Feature/pelagos 4373 create cleanup job

### PR DESCRIPTION
**Pull request by mickel1138**

### CODE REVIEW

File: src/MessageHandler/DeleteFileHandler.php [hunk 1]
```diff
@@ -45,13 +56,33 @@ public function __construct(LoggerInterface $deleteFileLogger, Datastore $datast
...
     {
-        $filePath = $deleteFile->getFilePath();
-        $this->logger->info(sprintf('Processing File with ID: "%s"', $filePath));
-        try {
-            $this->datastore->deleteFile($filePath);
-        } catch (\Exception $e) {
-            $this->logger->error(sprintf('Unable to delete file. Message: "%s"', $e->getMessage()));
+        $datasetSubmissionId = $deleteFile->getDatasetSubmissionId();
+
+        $this->logger->info(sprintf('Processing File with ID: "%s"', $datasetSubmissionId));
```
Comment (mickel1138) [inline src/MessageHandler/DeleteFileHandler.php:61]: Need better message!

File: src/MessageHandler/DeleteFileHandler.php [hunk 2]
```diff
@@ -45,13 +56,33 @@ public function __construct(LoggerInterface $deleteFileLogger, Datastore $datast
...
+
+        if (!$datasetSubmission instanceof DatasetSubmission) {
+            $this->logger->warning('Submission not found');
             return;
         }
+
+        $fileset = $datasetSubmission->getFileset();
+
+        if (!$fileset instanceof Fileset) {
+            $this->logger->warning('fileset not found');
```
Comment (mickel1138) [inline src/MessageHandler/DeleteFileHandler.php:73]: Need better message!

File: src/MessageHandler/DeleteFileHandler.php [hunk 3]
```diff
@@ -45,13 +56,33 @@ public function __construct(LoggerInterface $deleteFileLogger, Datastore $datast
...
-        } catch (\Exception $e) {
-            $this->logger->error(sprintf('Unable to delete file. Message: "%s"', $e->getMessage()));
+        $datasetSubmissionId = $deleteFile->getDatasetSubmissionId();
+
+        $this->logger->info(sprintf('Processing File with ID: "%s"', $datasetSubmissionId));
+
+        $datasetSubmission = $this->entityManager->getRepository(DatasetSubmission::class)->find($datasetSubmissionId);
+
+        if (!$datasetSubmission instanceof DatasetSubmission) {
+            $this->logger->warning('Submission not found');
```
Comment (mickel1138) [inline src/MessageHandler/DeleteFileHandler.php:66]: Need better message!

File: src/MessageHandler/DeleteFileHandler.php [hunk 4]
```diff
@@ -45,13 +56,33 @@ public function __construct(LoggerInterface $deleteFileLogger, Datastore $datast
...
+        foreach ($fileset->getDeletedFiles() as $deletedFile) {
+            try {
+                $this->datastore->deleteFile($deletedFile->getPhysicalFilePath());
+            } catch (\Exception $e) {
+                $this->logger->error(sprintf('Unable to delete file. Message: "%s"', $e->getMessage()));
+            }
+            $fileset->removeFile($deletedFile);
+        }
+
+        $this->entityManager->flush($datasetSubmission);
```
Comment (praneethpr) [inline src/MessageHandler/DeleteFileHandler.php:86]: Do we need this to be passed?


From the Doctrine/ORM/EntityManager.php, flush() method comment 

>      If an entity is explicitly passed to this method only this entity and the cascade-persist semantics + scheduled inserts/removals are synchronized.

**Review by praneethpr — APPROVED** [review]
Tested and works

**Review by fightingtexasaggie — APPROVED** [review]
reviewed, tested, working.

**Pull request merged**